### ISSUE TITLE
Add support for Azure Spot VMSS

### ISSUE THREAD

**Issue by CecileRobertMichon** [kubernetes-sigs/cluster-api-provider-azure#575]
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/use-spot

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

See #407 and #559 for the VM implementation.

This is dependent on #483 merging.

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by gdhuper** [comment]
If this issue is still available, I'd like to take a stab at it. Thanks

**Comment by dr0pdb** [comment]
@CecileRobertMichon Might be a naive question but what is the difference between [`AzureMachine`](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/config/crd/bases/infrastructure.cluster.x-k8s.io_azuremachines.yaml) and [`AzureMachineTemplate`](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/config/crd/bases/infrastructure.cluster.x-k8s.io_azuremachinetemplates.yaml)? Both seem very similar.

**Comment by devigned** [comment]
@dr0pdb if you were able to see the types side by side, you'd see differences between them. The `AzureMachineTemplate` in the experimental folder is used in virtual machine scale sets and doesn't have all of the information that can be set on an Azure virtual machine template (providerID, tags, etc). In machine pools, where the experimental `AzureMachineTemplate` is used, the `AzureMachinePool` and other `MachinePool` implementations have providerID and providerIDList on the machine pool rather than on each machine since the machine pool (virtual machine scale set) is responsible for orchestration of VMs rather than CAPZ operator. 

In hindsight, I think the `AzureMachineTemplate` type in experimental should be renamed to assuage confusion and make it easier to incorporate the types in the experimental folder into the stable types.

wdyt?

**Comment by dr0pdb** [comment]
Hi @devigned , thanks for the clarification.

Yes, I agree with your point. It would help eradicate confusion. Shall I scope it in this issue or a separate issue for renaming?

**Comment by devigned** [comment]
I vote for keeping it as a separate issue.

**Comment by CecileRobertMichon** [comment]
+1 to separate issue, let's do any API changes as part of v1alpha4 (even it's experimental this would be a breaking change).
